### DISCUSSION TITLE
Power button always ON

### DISCUSSION THREAD

**Discussion by Glodson** [thoukydides/homebridge-homeconnect#260] [Q&A]
Hi all
Thanks for that great plugin. It’s an amazing support for my smart environment at home.  

Just one question. My Bosch washing machine and Bosch dryer are sudden always ON in my hombridge for some time. Unfortunately, I cannot answer, when it happened the first time. Is there an issue which is known already? I did not find anything until now. 

I install updates always immediate. All other functionalities/services (e.g. door opened and so on) are signalized absolutely accurate (also in homekit). I just talk about power signal, which is incorrect.

Homeconnect app works correct and absolutely fine as well. Means, if a machine is off it is also off in the app. 

Thanks!

**Comment by thoukydides** [comment]
This was discussed in #251.

It was an intentional change in [v0.40.1](https://github.com/thoukydides/homebridge-homeconnect/compare/v0.40.0...v0.40.1).

My understanding is that the power switch on (some?) Washer/Dryer appliances switches everything off including the Home Connect functionality. This results in the Home Connect servers just losing the connection with the appliance, without any indication as to the cause, so after a timeout they generate a `DISCONNECTED` event. There is no way to distinguish between the appliance being switched off and any of the many other reasons why the connection to the appliance has been lost; other than being disconnected it is not possible to read any of the other appliance status such as whether it is switched on.

None of my Home Connect appliances behave like that - they all have soft power buttons that leave the Home Connect functionality active, so they indicate that they are Off or in Standby.

Earlier versions of this plugin set a default inactive HomeKit state whenever it was not possible to determine the actual appliance state. In v0.40.1 I changed this to set the Power state to `SERVICE_COMMUNICATION_FAILURE` instead of OFF to more accurately reflect the Home Connect semantics.

**Comment by Glodson** [comment]
It just only works with an rollback to 0.40.0. All other higher versions produce the same issue as above. Means, power switch gets on after first run of these machines (washing machine or dryer) and stay on this level (switch is on)

**Comment by thoukydides** [comment]
Like I said, this was a deliberate change of behaviour in v0.40.1 to more accurate reflect the information (or rather the lack thereof) from the Home Connect API. When the appliance disconnects from Home Connect (possibly due to being switched off but not necessarily), this plugin now sets a HomeKit `SERVICE_COMMUNICATION_FAILURE` error state instead of an arbitrary Power state.
